apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineElement.jsx

```diff
@@ -0,0 +1,181 @@
+// 1. CREATED - this is the node that represents the first element on the timeline, it has a line extending below
+//    it unless it is also the only element in the timeline. Otherwise this node does not vary from project to project.
+// 2. COMMIT - this represents a commit to the project. It will display an eyeball on the left which links to the
+//    project version if the project version is present and not expired. It also handles a child which represents
+//    the specific content in the commit. If it is the last element in the timeline, it will not have a line extending
+//    down from it.
+// 3. CODE_REVIEW - this represnts a code review request and comments made. It will display an eyeball to the left
+//    which links to the project version if it is present and not expired. It takes a child which should be the actual
+//    code review component. If it is the last element in the timeline it will not have a line extending below it.
+const CodeReviewTimelineElement = ({
+  type,
+  isLast,
+  projectVersionId,
+  isProjectVersionExpired,
+  children
+}) => {
+  const displayVersion = !isProjectVersionExpired && !!projectVersionId;
+  const versionLink =
+    location.origin + location.pathname + '?version=' + projectVersionId;
+
+  if (type === codeReviewTimelineElementType.CREATED) {
+    return (
+      <div style={styles.element}>
+        <div style={styles.eyeColumn} />
+        <div style={styles.timeline}>
+          <TimelineDot color={color.purple} />
+          {!isLast && <TimelineLine height={40} marginRight={-1.5} />}
+        </div>
+        <div style={{...styles.child, ...styles.createdText}}>
+          {javalabMsg.created()}
+        </div>
+      </div>
+    );
+  }
+
+  if (type === codeReviewTimelineElementType.COMMIT) {
+    const borderLeft = isLast ? 'none' : lineStyle;
+
+    return (
+      <div style={styles.element}>
+        <div style={styles.eyeColumn}>
+          {displayVersion && <EyeballLink versionHref={versionLink} />}
+        </div>
+        <div style={styles.timeline}>
+          <TimelineDot color={color.dark_charcoal} hasCheck={true} />
+        </div>
+        <div style={{...styles.commitChild, borderLeft}}>{children}</div>
+      </div>
+    );
+  }
+
+  if (type === codeReviewTimelineElementType.CODE_REVIEW) {
+    return (
+      <div style={styles.element}>
+        <div style={styles.eyeColumn}>
+          {displayVersion && <EyeballLink versionHref={versionLink} />}
+        </div>
+        <div style={styles.codeReviewTimeline}>
+          <div>{children}</div>
+          {!isLast && <TimelineLine height={40} marginLeft={15} />}
+        </div>
+      </div>
+    );
+  }
+};
+
+CodeReviewTimelineElement.propTypes = {
+  type: PropTypes.oneOf(Object.values(codeReviewTimelineElementType))
+    .isRequired,
+  isLast: PropTypes.bool,
+  projectVersionId: PropTypes.string,
+  isProjectVersionExpired: PropTypes.bool,
+  children: PropTypes.node
+};
+
+// Helper to render the eyeball
```

Comment:
I like all these helpers! It really helps to extract some of the logic, particularly around styles

Reply:
The thing that these were actually the most useful for was writing tests and verifying that the eyeball or line were rendered